src/client/tagClosing.ts

```diff
@@ -14,87 +14,73 @@ export interface AutoCloseResult {
-		}
-		let lastChange = changes[changes.length - 1];
-		let lastCharacter = lastChange.text[lastChange.text.length - 1];
-		if (lastChange.rangeLength > 0 || lastChange.text.length > 1 || TRIGGER_CHARACTERS.indexOf(lastCharacter) < 0) {
-			return;
-		}
-		let rangeStart = lastChange.range.start;
-		let version = document.version;
-		timeout = setTimeout(() => {
-			let position = new Position(rangeStart.line, rangeStart.character + lastChange.text.length);
-			tagProvider(document, position).then(result => {
-				if (!result) {
-					return;
-				}
-				let text = result.snippet;
-				let replaceLocation : Position | Range;
-				let range : Range = result.range;
-				if(range != null) {
-					// re-create Range
-					let line = range.start.line;
-					let character = range.start.character;
-					let startPosition = new Position(line, character);
-					line = range.end.line;
-					character = range.end.character;
-					let endPosition = new Position(line, character);
-					replaceLocation = new Range(startPosition, endPosition);
-				}
-				else {
-					replaceLocation = position;
-				}
-				if (text && isEnabled) {
-					let activeEditor = window.activeTextEditor;
-					if (activeEditor) {
-						let activeDocument = activeEditor.document;
-						if (document === activeDocument && activeDocument.version === version) {
-							activeEditor.insertSnippet(new SnippetString(text), replaceLocation);
-							
-						}
-					}
-				}
-			}, (reason: any) => {
-				console.log('xml/closeTag request has been cancelled');
-			});
-			timeout = void 0;
-		}, 100);
-	}
-	return Disposable.from(...disposables);
+  function updateEnabledState() {
+    isEnabled = false;
+    let editor = window.activeTextEditor;
+    if (!editor) {
+      return;
+    }
+    let document = editor.document;
+    if (!supportedLanguages[document.languageId]) {
+      return;
+    }
+    if (!workspace.getConfiguration(undefined, document.uri).get<boolean>(configName)) {
+      return;
+    }
+    isEnabled = true;
+  }
+
+  function onDidChangeTextDocument(document: TextDocument, changes: ReadonlyArray<TextDocumentContentChangeEvent>) {
+    if (!isEnabled) {
+      return;
+    }
+    let activeDocument = window.activeTextEditor && window.activeTextEditor.document;
+    if (document !== activeDocument || changes.length === 0) {
+      return;
+    }
+    if (typeof timeout !== 'undefined') {
+      clearTimeout(timeout);
+    }
+    let lastChange = changes[changes.length - 1];
+    let lastCharacter = lastChange.text[lastChange.text.length - 1];
+    if (lastChange.rangeLength > 0 || lastCharacter !== '>' && lastCharacter !== '/') {
+      return;
+    }
+    let rangeStart = lastChange.range.start;
+    let version = document.version;
+    timeout = setTimeout(() => {
+      let position = new Position(rangeStart.line, rangeStart.character + lastChange.text.length);
+      tagProvider(document, position).then(result => {
+        const text = result?.snippet;
```

Comment:
This means that the functionality described in https://github.com/redhat-developer/vscode-xml/issues/440 no longer works, since that functionality depends on using the returned range to determine what should be deleted.